scripts/freeze

```diff
@@ -0,0 +1,21 @@
+#!/bin/sh -e
+
+# Use the Python executable provided from the `-p` option, or a default.
+[ "$1" = "-p" ] && PYTHON=$2 || PYTHON="python3"
+
+VENV=`mktemp -d ./venv-XXXXXXXXXX`
+
+set -x
+
+"${PYTHON}" -m venv "${VENV}"
+PIP="${VENV}/bin/pip"
+
+"${PIP}" install -U "pip >= 20.2"
+"${PIP}" install -r requirements.base
+
+# Argument "--all" is needed to pin "wheel". Exclude our own editable dependency
+"${PIP}" freeze --all | grep -v '^-e' > requirements.txt
+# Add it back
+echo "-e .[http2]" >> requirements.txt
```

Comment:
Why not let `pip freeze` handle editable dependency?

Reply:
```console
$ pip install -e .[http2]
Obtaining file:///Users/josepcugat/workspace/httpcore
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting sniffio==1.*
  Using cached sniffio-1.1.0-py3-none-any.whl (4.5 kB)
Collecting h11<0.10,>=0.8
  Using cached h11-0.9.0-py2.py3-none-any.whl (53 kB)
Collecting h2==3.*; extra == "http2"
  Using cached h2-3.2.0-py2.py3-none-any.whl (65 kB)
Collecting hpack<4,>=3.0
  Using cached hpack-3.0.0-py2.py3-none-any.whl (38 kB)
Collecting hyperframe<6,>=5.2.0
  Using cached hyperframe-5.2.0-py2.py3-none-any.whl (12 kB)
Installing collected packages: sniffio, h11, hpack, hyperframe, h2, httpcore
  Running setup.py develop for httpcore
Successfully installed h11-0.9.0 h2-3.2.0 hpack-3.0.0 httpcore hyperframe-5.2.0 sniffio-1.1.0
$ pip freeze
h11==0.9.0
h2==3.2.0
hpack==3.0.0
-e git+git@github.com:encode/httpcore.git@26a37dc1d0f6961bb71ce18331f3c3b45c6f928b#egg=httpcore
hyperframe==5.2.0
sniffio==1.1.0
```

It would be pinning `httpcore` to a specific commit of the git repo instead of the current dir.

Reply:
How about `pip freeze --exclude-editable` ?

Reply:
Even better! Didn't know about this argument, updated 👍